src/librustc_front/hir.rs

```diff
@@ -195,6 +205,15 @@ pub struct PathSegment {
     pub parameters: PathParameters,
 }
 
+impl PathSegment {
+    pub fn new(ident: Ident, params: PathParameters) -> PathSegment {
+        PathSegment {
+            identifier: ident,
+            parameters: params,
+        }
+    }
+}
+
```

Comment:
I would remove these new impls and construct the types directly (i.e. with struct expressions) in `resolve` to be consistent with how the types are constructed in the rest of the codebase.